src/domain/core/src/messages/core_message_types.rs

```diff
@@ -57,6 +76,21 @@ pub struct DatasetLifecycleMessageCreated {
     pub owner_account_id: AccountID,
     #[serde(default)]
     pub dataset_visibility: DatasetVisibility,
+    // Note: Since we already have messages stored in the database,
+    //       to avoid breaking serialization, we use Optional type
+    dataset_name: Option<DatasetName>,
+}
+
+impl DatasetLifecycleMessageCreated {
+    pub fn dataset_name(&self) -> Cow<DatasetName> {
+        if let Some(name) = &self.dataset_name {
+            Cow::Borrowed(name)
+        } else {
+            tracing::warn!("Dataset '{}' does not has a name", self.dataset_id);
+
+            Cow::Owned(DatasetName::new_unchecked("???"))
+        }
+    }
```

Comment:
I don't understand this. If name can be missing in the domain, it's a reason to use `Option`. If name is missing from technical reasons, it's an internal error.

Reply:
Discussed in Slack

Optionality has been removed

Done